mmf_cli/inference.py

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env python3 -u
+# Copyright (c) Facebook, Inc. and its affiliates.
+import argparse
+import typing
+
+from mmf.utils.flags import flags
+from mmf.utils.inference import Inference
+from omegaconf import OmegaConf
+
+
+def construct_config(opts: typing.List[str]):
+    config = OmegaConf.create()
+    opt_values = [opt.split("=", maxsplit=1) for opt in opts]
+    if not all(len(opt) == 2 for opt in opt_values):
+        for opt in opt_values:
+            assert len(opt) == 2, f"{opt} has no value"
+
+    for opt, value in opt_values:
+        config[opt] = value
+
+    return config
+
+
+def inference(opts: typing.Optional[typing.List[str]] = None):
+    """Inference runs inference on an image and text provided by the user.
+    You can optionally run inference programmatically by passing an optlist as opts.
+
+    Args:
+        opts (typing.Optional[typing.List[str]], optional): Optlist which can be used.
+            to override opts programmatically. For e.g. if you pass
+            opts = ["text='how many dogs'", "checkpoint_path=my/directory"],
+            this will set the checkpoint for the model and the text to pass to it.
+    """
+    if opts is None:
+        parser = flags.get_parser()
+        args = parser.parse_args()
+    else:
+        args = argparse.Namespace(config_override=None)
+        args.opts = opts
+
+    config = construct_config(args.opts)
+    inference = Inference(checkpoint_path=config.checkpoint_path)
```

Comment:
model is suggested as an input in your command, but where is it being used. how does inference figure out based on checkpoint which model it is?

Reply:
Model is suggested as we are looking to eventually make this work with the zoo, but for now inference figures out which model using the checkpoint's config